### DISCUSSION TITLE
Support for readPixel(int16_t x, int16_t y, uint16_t *color)

### DISCUSSION THREAD

**Discussion by dapostol73** [moononournation/Arduino_GFX#285] [Ideas]
It would be super helpful to have the ability to read the current value of a pixel, this would help with compositing gradient alpha images to the existing background.

**Comment by moononournation** [comment]
As mentioned in instructables, Arduino_GFX does not have read functions.

**Reply by dapostol73** [reply]
Is the API for this library restricted by the Arduino_GFX library define?

**Reply by moononournation** [reply]
Yes, you can consider other sophisticated GFX library.

**Reply by dapostol73** [reply]
Is there a way I can use the bus to do this myself?

**Reply by moononournation** [reply]
no

**Reply by moononournation** [reply]
You should read the instructables first to know the reason before ask further questions.

**Reply by dapostol73** [reply]
Okay, thanks

**Comment by h5n1xp** [comment]
> It would be super helpful to have the ability to read the current value of a pixel, this would help with compositing gradient alpha images to the existing background.

@dapostol73 
If you would be prepared to use my sprite library on top of Arduino_GFX, then I could add pixel reading to my sprite library if that would be helpful.

**Reply by dapostol73** [reply]
@h5n1xp do you have example documentation of how to use the Sprite library you are referring too?  And maybe a like to the repo, so I can have a look?

**Reply by h5n1xp** [reply]
At the moment it is just for fast sprites on screen, but I'm planning in add some extra features as I progress through the BETAs, I'm happy to include a readPixel() method for the display if that would-be helpful.

https://github.com/h5n1xp/Arduino_Sprite

**Reply by h5n1xp** [reply]
Hi @dapostol73, if you use my Arduino_Sprite library on top Arduino_GFX, then use Arduino_FrameBuffer in place of Arduino_Canvas, as the FrameBuffer class does have a readPixel(x,y) method... Otherwise it inherits from Arduio_Canvas so supports all the same methods/properties.
